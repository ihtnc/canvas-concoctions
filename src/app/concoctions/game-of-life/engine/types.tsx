import { type HSL, type RenderFunction } from "@/utilities/drawing-operations";
import { type MatrixOperationFunction, type MatrixValue } from "@/utilities/matrix-operations";

export enum CycleState {
  Start,
  Stop
}

export enum CellState {
  Alive,
  Dying,
  Dead,
  Growing
};
export type CellValue = { state: CellState };

export type CreateCellOperationFunction<T> = (cycleIndex: number, data?: T) => CellOperationFunction;

export interface CellOperationFunction extends MatrixOperationFunction {
  (value: MatrixValue<CellValue>): MatrixValue<CellValue>
};

export type RenderPipelineData = {
  map: MatrixValue<CellValue>,
  width: number,
  height: number,
  aliveColor: HSL,
  dyingColor: HSL,
  growingColor: HSL,
  cycleIndex: number
};

export interface CellRenderFunction extends RenderFunction {
  (context: CanvasRenderingContext2D, data: RenderPipelineData): void;
};
