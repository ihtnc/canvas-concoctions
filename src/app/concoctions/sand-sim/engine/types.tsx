import { type HSL } from "@/utilities/drawing-operations"
import { type MatrixCoordinate, type MatrixValue } from "@/utilities/matrix-operations"

export type PageData = {
  resizeMap: boolean,
  map: MatrixValue<ParticleValue>,
  resetMap: boolean,
  currentColor: HSL,
  canGenerateParticle: boolean,
  newParticleCoordinate?: MatrixCoordinate,
  particleWidth: number,
  particleHeight: number,
}

export type ParticleValue = { color?: HSL, value: number }

export type ParticleColorMap = { color: HSL, particles: Array<MatrixCoordinate> }