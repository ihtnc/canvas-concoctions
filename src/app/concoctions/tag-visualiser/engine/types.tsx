import { type RenderFunction, type Size } from "@/utilities/drawing-operations"
import { type OperationFunction } from "@/utilities/misc-operations"
import { type Transition, type TransitionProps } from "@/utilities/transition-operations"
import { type Coordinates } from "@/components/canvas/types"

export enum TagState {
  Normal,
  NewTag,
  NewProps
}

export type TagValue = {
  value: string,
  rank: number,
  count: number,
  props: TransitionProps,
  targetProps: TransitionProps,
  state: TagState,
  transitions: Array<Transition>
}

export type TagAllocations = {
  origin: Coordinates,
  allocations: {
    [tagKey: string]: {
      tag: TagValue,
      allocation: AllocatedSpace<string>
    }
  }
}

export type Tags = { [tag: string]: TagValue };

export type TagOperationData = {
  frame: number,
  tag: TagValue
}
export interface TagOperationFunction extends OperationFunction {
  (value: TagOperationData): TagOperationData
}

export type RenderPipelineData = {
  frame: number,
  tags: Tags
};

export interface TagsRenderFunction extends RenderFunction {
  (context: CanvasRenderingContext2D, data: RenderPipelineData): void;
}

export type PackedSpace<T> = {
  origin: Coordinates,
  size: Size,
  free: Array<Space>
  allocations: Array<AllocatedSpace<T>>
}

export type Space = {
  location: Coordinates,
  size: Size
}

export interface AllocatedSpace<T> extends Space {
  horizontal: boolean,
  value: T
}