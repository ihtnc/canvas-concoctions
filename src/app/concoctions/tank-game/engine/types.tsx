import { type Coordinates } from "@/components/canvas/types"
import { type Size, type RenderFunction, type RGB } from "@/utilities/drawing-operations"
import { type OperationFunction } from "@/utilities/misc-operations"

export type GameObject = {
  tank: TankObject,
  gunBarrel: GunBarrelObject,
  target: TargetObject,
  bullet: BulletObject,
  explosion: ExplosionObject,
  message: MessageObject,
  gameOver: GameOverObject,
  controls: ControlsObject
}

type TankObject = {
  location: Coordinates,
  angle: number,
  power: number,
  bullets: number
}

type GunBarrelObject = {
  location: Coordinates,
  rotation: Coordinates
}

type TargetObject = {
  location: Coordinates,
  origin: Coordinates,
  currentDirection?: TargetDirection,
  isReversing: boolean,
  isHit: boolean
}

export enum TargetDirection {
  Left,
  Right,
  Up,
  Down
}

type BulletObject = {
  location: Coordinates,
  velocity: Coordinates,
  active: boolean,
  startFrame: number
}

type ExplosionObject = {
  location: Coordinates,
  active: boolean,
  startFrame: number
}

type MessageObject = {
  active: boolean,
  hit: boolean,
  startFrame: number
}

type GameOverTextObject = {
  text: string,
  location: Coordinates,
  size: Size
}

type GameOverObject = {
  active: boolean,
  startFrame: number,
  message: GameOverTextObject,
  score: GameOverTextObject,
  highScore: GameOverTextObject,
  newHighScore: boolean
}

type ControlsObject = {
  powerUp: ControlObject,
  powerDown: ControlObject,
  angleUp: ControlObject,
  angleDown: ControlObject,
  gunBarrelFire: ControlObject
  fire: ControlObject,
  restart: ControlObject
}

type ControlObject = {
  location: Coordinates
}

export enum State {
  Ready,
  TurnComplete,
  GameOver
}

export enum Command {
  PowerUp,
  PowerDown,
  AngleUp,
  AngleDown,
  Fire,
  Restart
}

export enum Difficulty {
  Normal,
  RepositionTarget,
  MoveTargetTwoWay,
  MoveTargetFourWay,
  MoveTargetRandomly
}

export type GameStateObject = {
  frame: number,
  score: number,
  hiScore: number
  difficulty: number,
  totalHits: number,
  size: Size,
  currentCommand?: Command,
  state: State
}

export type GameConfig = {
  tank: Readonly<TankConfig>,
  gunBarrel: Readonly<GunBarrelConfig>,
  target: Readonly<TargetConfig>,
  bullet: Readonly<BulletConfig>,
  rank: Readonly<RankConfig>,
  trajectory: Readonly<TrajectoryConfig>,
  explosion: Readonly<ExplosionConfig>,
  environment: Readonly<EnvironmentConfig>,
  controls: Readonly<ControlsConfig>,
  stats: Readonly<StatsConfig>,
  message: Readonly<MessageConfig>,
  gameOver: Readonly<GameOverConfig>,
  localStorage: Readonly<LocalStorageConfig>
}

type TankConfig = {
  size: Readonly<Size>
}

type GunBarrelConfig = {
  offset: Readonly<Coordinates>,
  size: Readonly<Size>,
  angleMultiplier: Readonly<number>
}

type TargetConfig = {
  size: Readonly<Size>,
  movementRange: Readonly<number>,
  movementSpeedMultiplier: Readonly<number>
}

type BulletConfig = {
  offset: Readonly<Coordinates>,
  size: Readonly<Size>,
  animationSpeed: Readonly<number>,
  padding: Readonly<number>
}

type RankConfig = {
  offset: Readonly<Coordinates>,
  size: Readonly<Size>
}

type TrajectoryConfig = {
  lineWidthMultiplier: Readonly<number>,
  length: Readonly<number>,
  lineDash: ReadonlyArray<number>
}

type ExplosionConfig = {
  size: Readonly<Size>,
  duration: Readonly<number>
}

type EnvironmentConfig = {
  gravity: Readonly<number>,
  airResistance: Readonly<number>,
  airResistanceModifier: Readonly<number>,
  powerMultiplier: Readonly<number>,
  maxAngle: Readonly<number>,
  minAngle: Readonly<number>,
  maxPower: Readonly<number>,
  minPower: Readonly<number>,
  hitsPerDifficulty: Readonly<number>
}

type ControlsConfig = {
  size: Readonly<Size>,
  padding: Readonly<number>,
  activeOpacity: Readonly<number>,
  inactiveOpacity: Readonly<number>,
  powerUp: Readonly<ControlConfig>,
  powerDown: Readonly<ControlConfig>,
  angleUp: Readonly<ControlConfig>,
  angleDown: Readonly<ControlConfig>,
  gunBarrelFire: Readonly<ControlConfig>,
  fire: Readonly<ControlConfig>,
  restart: Readonly<ControlConfig>
}

type ControlConfig = {
  offset: Readonly<Coordinates>,
  sizeMultiplier: Readonly<number>
}

type StatsConfig = {
  location: Readonly<Coordinates>,
  padding: Readonly<number>,
  color: Readonly<RGB>,
  font: Readonly<string>
}

type MessageTextConfig = {
  color: Readonly<RGB>,
  font: Readonly<string>
}

type MessageConfig = {
  duration: Readonly<number>,
  hit: Readonly<MessageTextConfig>,
  miss: Readonly<MessageTextConfig>
}

type GameOverConfig = {
  animationSpeed: Readonly<number>,
  padding: Readonly<number>,
  message: Readonly<MessageTextConfig>,
  score: Readonly<MessageTextConfig>,
  highScore: Readonly<MessageTextConfig>
}

type LocalStorageConfig = {
  highScoreKey: Readonly<string>
}

export type ResourcesObject = {
  tankImage: HTMLImageElement,
  gunBarrelImage: HTMLImageElement,
  targetImage: HTMLImageElement,
  bulletImage: HTMLImageElement,
  explosionImage: HTMLImageElement,
  arrowImage: HTMLImageElement,
  shootImage: HTMLImageElement,
  rankImage: HTMLImageElement,
  restartImage: HTMLImageElement
}

export type GameOperationData = {
  game: GameObject,
  state: GameStateObject
}
export interface GameOperationFunction extends OperationFunction {
  (value: GameOperationData): GameOperationData
}

export type RenderPipelineData = {
  frame: number,
  game: GameObject,
  state: GameStateObject,
  config: GameConfig,
  resources: ResourcesObject
}

export interface TankGameRenderFunction extends RenderFunction {
  (context: CanvasRenderingContext2D, data: RenderPipelineData): void;
}
