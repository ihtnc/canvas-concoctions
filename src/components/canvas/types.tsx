export type InitRenderHandler = (canvas: HTMLCanvasElement) => void;
export type ShouldRedrawHandler = (canvas: HTMLCanvasElement, context: CanvasRenderingContext2D) => boolean;
export type DrawHandler = (context: CanvasRenderingContext2D) => void;
export type PreDrawHandler = (canvas: HTMLCanvasElement, context: CanvasRenderingContext2D) => void;
export type PostDrawHandler = () => void;
export type DebugLayerDrawHandler = (value: DebugLayerValue, context: CanvasRenderingContext2D) => void;

export enum RenderLocation {
  TopLeft, TopCenter, TopRight,
  MiddleLeft, Center, MiddleRight,
  BottomLeft, BottomCenter, BottomRight
};

export type Coordinates = { x: number, y: number };
export type DebugLayerOptions = {
  location: RenderLocation | Coordinates,
  renderFps: boolean,
  renderSize: boolean,
  renderClientSize: boolean,
  renderPixelRatio: boolean
};

export type DebugLayerValue = {
  fps: number,
  width: number,
  height: number,
  clientWidth: number,
  clientHeight: number,
  pixelRatio: number
}

export type DebugLayerRendererValue = boolean | RenderLocation | Coordinates | DebugLayerOptions | DebugLayerDrawHandler;

export type Use2DRenderLoopOptions = {
  clearEachFrame?: boolean,
  onInit?: InitRenderHandler,
  onDraw?: DrawHandler,
  onPreDraw?: PreDrawHandler,
  onPostDraw?: PostDrawHandler,
  onShouldRedraw?: ShouldRedrawHandler,
  debugLayerRenderer?: DebugLayerRendererValue
}

export type OnResizeHandler = (canvas: HTMLCanvasElement, width: number, height: number) => void;