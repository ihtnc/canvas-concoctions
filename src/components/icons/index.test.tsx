import { beforeAll, describe, expect, test } from "vitest"
import {
  ForwardIcon,
  PauseIcon,
  PlayIcon,
  TrashIcon,
  TagIcon,
  PlusCircleIcon,
  BeakerIcon
} from "."
import { render } from '@testing-library/react'
import { JSXElementConstructor } from "react"
import { IconProps } from "./types"

describe('icons', () => {
  const iconList: Array<{Icon: JSXElementConstructor<IconProps>, title: string}> = [
    { Icon: ForwardIcon, title: 'ForwardIcon' },
    { Icon: PauseIcon, title: 'PauseIcon' },
    { Icon: PlayIcon, title: 'PlayIcon' },
    { Icon: TrashIcon, title: 'TrashIcon' },
    { Icon: TagIcon, title: 'TrashIcon' },
    { Icon: PlusCircleIcon, title: 'PlusCircleIcon' },
    { Icon: BeakerIcon, title: 'BeakerIcon' }
  ]

  test.each(iconList)('should define $title', ({ Icon }: { Icon: JSXElementConstructor<IconProps> }) => {
    const { container } = render(<Icon />)
    expect(container).toMatchSnapshot()
  })

  test.each(iconList)('should set className ($title)', ({ Icon }: { Icon: JSXElementConstructor<IconProps> }) => {
    const className = "test-class-name"
    const { container } = render(<Icon className={className} />)
    expect(container).toMatchSnapshot()
  })
})