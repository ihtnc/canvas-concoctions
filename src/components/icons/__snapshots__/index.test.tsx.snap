// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`icons > should define 'BeakerIcon' 1`] = `
<div>
  <svg
    class="w-6 h-6 "
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9.75 3.104v5.714a2.25 2.25 0 0 1-.659 1.591L5 14.5M9.75 3.104c-.251.023-.501.05-.75.082m.75-.082a24.301 24.301 0 0 1 4.5 0m0 0v5.714c0 .597.237 1.17.659 1.591L19.8 15.3M14.25 3.104c.251.023.501.05.75.082M19.8 15.3l-1.57.393A9.065 9.065 0 0 1 12 15a9.065 9.065 0 0 0-6.23-.693L5 14.5m14.8.8 1.402 1.402c1.232 1.232.65 3.318-1.067 3.611A48.309 48.309 0 0 1 12 21c-2.773 0-5.491-.235-8.135-.687-1.718-.293-2.3-2.379-1.067-3.61L5 14.5"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`icons > should define 'ForwardIcon' 1`] = `
<div>
  <svg
    class="w-6 h-6 "
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3 8.689c0-.864.933-1.406 1.683-.977l7.108 4.061a1.125 1.125 0 0 1 0 1.954l-7.108 4.061A1.125 1.125 0 0 1 3 16.811V8.69ZM12.75 8.689c0-.864.933-1.406 1.683-.977l7.108 4.061a1.125 1.125 0 0 1 0 1.954l-7.108 4.061a1.125 1.125 0 0 1-1.683-.977V8.69Z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`icons > should define 'PauseIcon' 1`] = `
<div>
  <svg
    class="w-6 h-6 "
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.75 5.25v13.5m-7.5-13.5v13.5"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`icons > should define 'PlayIcon' 1`] = `
<div>
  <svg
    class="w-6 h-6 "
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M5.25 5.653c0-.856.917-1.398 1.667-.986l11.54 6.347a1.125 1.125 0 0 1 0 1.972l-11.54 6.347a1.125 1.125 0 0 1-1.667-.986V5.653Z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`icons > should define 'PlusCircleIcon' 1`] = `
<div>
  <svg
    class="w-6 h-6 "
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 9v6m3-3H9m12 0a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`icons > should define 'TrashIcon' 1`] = `
<div>
  <svg
    class="w-6 h-6 "
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`icons > should define 'TrashIcon' 2`] = `
<div>
  <svg
    class="w-6 h-6 "
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9.568 3H5.25A2.25 2.25 0 0 0 3 5.25v4.318c0 .597.237 1.17.659 1.591l9.581 9.581c.699.699 1.78.872 2.607.33a18.095 18.095 0 0 0 5.223-5.223c.542-.827.369-1.908-.33-2.607L11.16 3.66A2.25 2.25 0 0 0 9.568 3Z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
    <path
      d="M6 6h.008v.008H6V6Z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`icons > should set className ('BeakerIcon') 1`] = `
<div>
  <svg
    class="w-6 h-6 test-class-name"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9.75 3.104v5.714a2.25 2.25 0 0 1-.659 1.591L5 14.5M9.75 3.104c-.251.023-.501.05-.75.082m.75-.082a24.301 24.301 0 0 1 4.5 0m0 0v5.714c0 .597.237 1.17.659 1.591L19.8 15.3M14.25 3.104c.251.023.501.05.75.082M19.8 15.3l-1.57.393A9.065 9.065 0 0 1 12 15a9.065 9.065 0 0 0-6.23-.693L5 14.5m14.8.8 1.402 1.402c1.232 1.232.65 3.318-1.067 3.611A48.309 48.309 0 0 1 12 21c-2.773 0-5.491-.235-8.135-.687-1.718-.293-2.3-2.379-1.067-3.61L5 14.5"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`icons > should set className ('ForwardIcon') 1`] = `
<div>
  <svg
    class="w-6 h-6 test-class-name"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M3 8.689c0-.864.933-1.406 1.683-.977l7.108 4.061a1.125 1.125 0 0 1 0 1.954l-7.108 4.061A1.125 1.125 0 0 1 3 16.811V8.69ZM12.75 8.689c0-.864.933-1.406 1.683-.977l7.108 4.061a1.125 1.125 0 0 1 0 1.954l-7.108 4.061a1.125 1.125 0 0 1-1.683-.977V8.69Z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`icons > should set className ('PauseIcon') 1`] = `
<div>
  <svg
    class="w-6 h-6 test-class-name"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M15.75 5.25v13.5m-7.5-13.5v13.5"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`icons > should set className ('PlayIcon') 1`] = `
<div>
  <svg
    class="w-6 h-6 test-class-name"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M5.25 5.653c0-.856.917-1.398 1.667-.986l11.54 6.347a1.125 1.125 0 0 1 0 1.972l-11.54 6.347a1.125 1.125 0 0 1-1.667-.986V5.653Z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`icons > should set className ('PlusCircleIcon') 1`] = `
<div>
  <svg
    class="w-6 h-6 test-class-name"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 9v6m3-3H9m12 0a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`icons > should set className ('TrashIcon') 1`] = `
<div>
  <svg
    class="w-6 h-6 test-class-name"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`icons > should set className ('TrashIcon') 2`] = `
<div>
  <svg
    class="w-6 h-6 test-class-name"
    fill="none"
    stroke="currentColor"
    stroke-width="1.5"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9.568 3H5.25A2.25 2.25 0 0 0 3 5.25v4.318c0 .597.237 1.17.659 1.591l9.581 9.581c.699.699 1.78.872 2.607.33a18.095 18.095 0 0 0 5.223-5.223c.542-.827.369-1.908-.33-2.607L11.16 3.66A2.25 2.25 0 0 0 9.568 3Z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
    <path
      d="M6 6h.008v.008H6V6Z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;
